source/agora/node/FullNode.d

```diff
@@ -491,6 +491,32 @@ public class FullNode : API
         if (this.pool.hasTransactionHash(tx_hash))
             return;
 
+        // all inputs have to be known in order to calculate the fee and
+        // compare it to the fee threshold
+        UTXO utxo;
+        foreach (const ref input; tx.inputs)
+            if (!utxo_set.peekUTXO(input.utxo, utxo))
+                return;
+
+        // only consider a double spend transaction, if its fee is
+        // considerably higher than the current highes fee
+        auto fee_threshold = ledger.getDoubleSpentHighestFee(tx);
+
+        // if the fee_threshold is null, it means there won't be any double
+        // spend transactions, after this transaction is added to the pool
+        if (!fee_threshold.isNull())
+        {
+            fee_threshold.div(100);
+            fee_threshold.mul(100 + config.node.double_spent_threshold_pct);
+        }
+
+        // an attacker could easily send a transaction with a very high fee
+        // which would make the fee_threshold invalid after the multiplication
+        // because of this, there is an explicit check to fee_threshold.isValid()
+        if (!fee_threshold.isNull() &&
+           (!fee_threshold.isValid() || fee_man.getTXFee(tx, &utxo_set.peekUTXO) < fee_threshold))
+            return;
```

Comment:
Do we think this is the right place to do this? Shouldn't this all be abstracted away in Ledger/TransactionPool?

Reply:
I agree, I was also thinking of moving this into a separate method in the Ledger class so it can be reused.